lib/awesome_spawn.rb

```diff
@@ -64,15 +64,12 @@ def run(command, options = {})
       options[:stdin_data] = in_data
     end
 
-    output, error, status = "", "", nil
     command_line = build_command_line(command, params)
 
     output, error, status = launch(command_line, options)
-  rescue Errno::ENOENT => err
-    raise NoSuchFileError.new(err.message) if NoSuchFileError.detected?(err.message)
```

Comment:
@Fryguy I went through linux, C (non ruby) and ruby docs. This is a linux/c error that is exposed through ruby.

I was unable to generate an error for this that was not a "NoSuchFileError". The code was untested and looked to be unneeded complexity.

I will try again and this one down. I believe there are other errors for bad symlink.

I'll probably just split this PR into 2.